common/initializer.go

```diff
@@ -339,11 +349,17 @@ func (i *Initializer) State() InitializationState {
 		Int("running", running).
 		Int("succeeded", succeeded).
 		Int("failed", failed).
+		Int("fatal", fatal).
 		Msg("calculating initialization state")
 
 	if total == succeeded {
 		return StateReady
 	}
+
+	if fatal > 0 {
+		return StateFatal
+	}
```

Comment:
Don't think we want to mark the whole initializer as Fatal if only 1 task is Fatal... I'd suggest to consider it like Partial if others are successful

Imagine from 10 upstreams 2 are fatally broken, but others are fine, then the Network initializer must be considered "Partial"